alacritty/src/config/monitor.rs

```diff
@@ -62,32 +64,68 @@ pub fn watch(mut paths: Vec<PathBuf>, event_proxy: EventLoopProxy<Event>) {
         parents.sort_unstable();
         parents.dedup();
 
-        let watcher = debouncer.watcher();
         // Watch all configuration file directories.
         for parent in &parents {
             if let Err(err) = watcher.watch(parent, RecursiveMode::NonRecursive) {
                 debug!("Unable to watch config directory {:?}: {}", parent, err);
             }
         }
 
+        // The current debouncing time.
+        let mut debouncing_time = None;
+
+        // The events accumulated during the debounce period.
+        let mut received_events = Vec::with_capacity(8);
```

Comment:
Is there any point in initializing this to a size of 8? Seems somewhat arbitrary and likely won't make any difference in practice?

Reply:
I just picked a value that will avoid any allocation.

Reply:
If anything it would avoid re-allocation, right? And the value is arbitrary, isn't it? Like it could easily be bigger than 8.

Reply:
The amount of events is `4` at max from what I've tested. Since you could have write/rename things. You can print the amount of events we're getting on writing to the file or replacing the file.

Reply:
That number seems arbitrary though? It will be completely different on other platforms.